src/nmea.c

```diff
@@ -167,44 +174,44 @@ void nmea_gpgga(const double pos_llh[3], const gps_time_t *gps_t, u8 n_used,
 
   unix_t = gps2time(gps_t);
   gmtime_r(&unix_t, &t);
-
-  double frac_s = fmod(gps_t->tow, 1.0);
-
-  s16 lat_deg = R2D * (pos_llh[0]);
-  double lat_min = MINUTES(pos_llh[0]);
-  s16 lon_deg = R2D * (pos_llh[1]);
-  double lon_min = MINUTES(pos_llh[1]);
-  lat_deg = abs(lat_deg);
-  lon_deg = abs(lon_deg);
-  char lat_dir = pos_llh[0] < 0 ? 'S' : 'N';
-  char lon_dir = pos_llh[1] < 0 ? 'W' : 'E';
-
-  NMEA_SENTENCE_START(120);
-  /* Note, geoid separation is set to 0 because no geoid model exists*/
+ 
+  double frac_s  = fmod(gps_t->tow, 1.0);
+  
+  double lat     = fabs( round( R2D * pos_llh[0] * 1e8 ) / 1e8 );
```

Comment:
In the next round. In fact all NMEA values should be computed once at the beginning of NMEA creation block and then used for all messages as needed. Code will be simpler, data will be consistent by design and run faster.